inc/class-tachyon.php

```diff
@@ -111,6 +111,24 @@ public static function parse_dimensions_from_filename( $src ) {
 		return array( false, false );
 	}
 
+	/**
+	 * Get an attachment post ID by GUID.
+	 *
+	 * @param string $src The attachment source.
+	 * @return int|false
+	 */
+	public static function get_attachment_by_guid( $src ) {
+		global $wpdb;
+
+		// Trim the host name portion of the source as this could potentially have changed.
+		$src_path = wp_parse_url( $src, PHP_URL_PATH );
+
+		// Try to fetch the post ID by GUID.
+		$id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM {$wpdb->posts} WHERE post_type = 'attachment' AND guid LIKE '%%%s' LIMIT 1;", $src_path ) );
```

Comment:
Does this need to be a like query? AFAIK `$src` should be the full URL? can we just match against that?

I think a `LIKE` on GUID is going to be too poor for performance to take this route.

Reply:
Yeah we can do, I just opted for path matching instead as the URL in the content might not match whatever the original GUID was. Could potentially look for `_wp_attached_file` in post meta for a direct match but that will probably have a similar issue as it's a much bigger table.

Reply:
Hmm yeah good thinking / point. I think if we go GUID we should go with URL. I think _wp_attached_file is probably the closest thing we have to "correct" and we'll likely get a better hitrate on this case, but it's going to be a tablescan alteast on the amount of rows that is the number of attachments in the DB.

The alternative would be a `file_exists` on the original file, though I'm not sure if that's better or worse.

Do we know how common this case is? I'm thinking atleast any non-altis authored content will hit this condition, as g-berg or the Classic Editor will be inserting `-300x200.jpg` suffixed images, so I'm thinking this condition is actually hit quite a lot.

We could also add object cacing for a url -> ID map in to this function or somewhere else.

Reply:
Yeah you're right, it'll be hit continually. Seems there's not going to be a consistently performant way of handling this.

Closing in favour of writing a WP CLI migration command instead to update old upload file names.